pkg/agent/baker.go

```diff
@@ -1411,7 +1418,10 @@ root = "{{GetDataDir}}"{{- end}}
 [plugins."io.containerd.cri.v1.images".registry.headers]
   X-Meta-Source-Client = ["azure/aks"]
 
-[plugins."io.containerd.cri.v1.runtime".containerd]
+{{ if IsCDIEnabled }}[plugins."io.containerd.cri.v1.runtime"]
+  enable_cdi = true
```

Comment:
CDI is enabled by default in containerd 2.0, so the logic here will need to be inverted such that if CDI is not enabled, then the config needs to disable it. Maybe it's best to explicitly define `enable_cdi` in each case.

https://github.com/containerd/containerd/pull/9621